RevolveBot

portaloffreedom opened the pull request.

Commit:
Add skeleton of future unified robot generator

maartenvanhooftds committed:
Initiated revolve_module.py + update for revolve_bot.py

maartenvanhooftds committed:
Initiated revolve_module.py + update for revolve_bot.py

portaloffreedom committed:
Merge remote-tracking branch 'origin/master' into revolve_bot

karinemiras committed:
Support YAML format in RevolveBot

karinemiras committed:
Add unittest for RevolveBot

karinemiras committed:
loads brain from yaml

karinemiras committed:
write brain to yaml

portaloffreedom committed:
Merge remote-tracking branch 'origin/master' into revolve_bot

zenfiric commented:
Please remove `pygazebo` as a git submodule. They are hard to keep updated, mostly causing headaches.

portaloffreedom committed:
Improved yaml parser

portaloffreedom committed:
Added Touch sensor

portaloffreedom committed:
if no brain type, assume neural network brain

safsaf150 committed:
Working -v rendering robots in 2d

safsaf150 committed:
Delete img's

portaloffreedom committed:
Loading SDF work in progress

only basic creation works:
- only up to depth one
- all in one link (no axis)

safsaf150 committed:
added id and hinge color to drawing

portaloffreedom committed:
Loading SDF work in progress

- loading all links correct
- missing joints
- missing inertials and mass

portaloffreedom committed:
ops

safsaf150 committed:
Merge remote-tracking branch 'upstream/revolve_bot' into render_robots_2d

safsaf150 committed:
reformatted code - using children enum

safsaf150 committed:
change type checking to class type

portaloffreedom committed:
SDF generation, inertial and mass

- implemented inertial for SDF links
- implemented mass for SDF links

portaloffreedom committed:
minor SDF rendering improvements

- improved tag naming
- activehinge does not influence the slot chain any more
(it only has one free slot, so it's obvious without naming it)
- added some stub for joint creation

portaloffreedom committed:
Loading SDF finally works

loading morphology is working, loading brain parameters is still not present.

portaloffreedom committed:
Loading SDF work in progress

only basic creation works:
- only up to depth one
- all in one link (no axis)

portaloffreedom committed:
Bugfixes:

- partially present brain would crash the plugin
- not updating the name when moving in Hinge connections would create naming conflicts, disabling some robotic joints.
- removed deprecated sdfbuilder code

portaloffreedom committed:
RobotManager

modifying the manager and robot manager files to use the new revolveBot
(also added position as a parameter when inserting the robot)

safsaf150 committed:
Measurements

safsaf150 committed:
Delete images

portaloffreedom committed:
SDF now respects revolvebot colors

karinemiras committed:
class lsystem + alphabet + initialization operator + early development

karinemiras committed:
Merge branch 'revolve_bot' of https://github.com/ci-group/revolve into revolve_bot

karinemiras committed:
read genotype + add params to development

portaloffreedom committed:
fix collision for joint without a brick attached

portaloffreedom committed:
Merge branch 'revolve_bot' of github.com:ci-group/revolve into revolve_bot

safsaf150 committed:
format code, trim whitespaces

safsaf150 committed:
delete test file

safsaf150 committed:
Merge remote-tracking branch 'upstream/revolve_bot' into render_robots_2d

portaloffreedom committed:
Merge pull request #36 from safsaf150/render_robots_2d

Render robots 2d

portaloffreedom committed:
Added touch sensor

portaloffreedom committed:
Added pycairo requirements

portaloffreedom committed:
Added touch sensor (SDF sensor wise)

maartenvanhooftds committed:
Enable new sdf brain structure

portaloffreedom committed:
Merge pull request #37 from ci-group/cpp_new_sdf

Cpp new sdf

portaloffreedom committed:
Fix rlpower bug

karinemiras committed:
implemented update_substrate function

portaloffreedom committed:
Merge pull request #38 from ci-group/update_substrate

implemented update_substrate function

safsaf150 committed:
measurements of body

safsaf150 committed:
rendering brain of robot

safsaf150 committed:
Merge remote-tracking branch 'upstream/revolve_bot' into brain_drawings_and_measurements

safsaf150 committed:
added brain drawing library to requirements

portaloffreedom committed:
Merge pull request #39 from safsaf150/brain_drawings_and_measurements

Brain drawings

portaloffreedom committed:
SDF loading into the simulation is working for brains as well!

portaloffreedom committed:
Merge branch 'revolve_bot' of github.com:ci-group/revolve into revolve_bot

portaloffreedom committed:
Remove const for compilation problem

safsaf150 committed:
brain measurements

safsaf150 committed:
Merge remote-tracking branch 'upstream/revolve_bot' into brain_drawings_and_measurements

portaloffreedom committed:
Rotated default orientation of the touch sensor

The old revolve had the touch sensor rotated 90 degrees from the current implementation.
Having a different tangent solved the problem.

safsaf150 committed:
calling any function multiple times will produce same results in body measurements

portaloffreedom committed:
Improved RevolveBot

- Removed redundant data from object
- Improved error printing for drawing functions

portaloffreedom committed:
Merge pull request #40 from safsaf150/brain_drawings_and_measurements

Brain measurements

portaloffreedom committed:
Added tutorial3 and improved manager to remove same robot

it's easier to run multiple times the manager

also added rlpower-splines brain to example robots

portaloffreedom committed:
Fixed unit tests
